File: JavaPractice/src/BinarySearch.java
import java.util.Arrays;


public class BinarySearch {
	
	public static void main(String arg[])
	{
		int[] a = {1,8,4,5,6,7,16,10,11,13,15};
		int k = 11;
		Arrays.sort(a);
		System.out.println("output is "+ isFind(a,k));
		
		
		
	}
	
public static boolean isFind(int[] a, int k)
{
	int low=0;
	int high=a.length-1;
	
	while(low<=high)
	{
		int mid=(low+high)/2;
		if(a[mid]==k)
		{
			return true;
		}
		else if(a[mid]<k)
		{
			low=mid+1;	
		}
		else if(a[mid]>k)
		{
			high=mid-1;
		}
	}
	return false;
}

}

File: PythonPractice/src/BinarySearch.py
from typing import List

def is_find(arr: List[int], k: int) -> bool:
    if not isinstance(arr, list) or not all(isinstance(x, int) for x in arr):
        raise TypeError("The first argument must be a list of integers")
    if not isinstance(k, int):
        raise TypeError("The second argument must be an integer")

    # Sorting the array
    arr.sort()

    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == k:
            return True
        elif arr[mid] < k:
            low = mid + 1
        else:
            high = mid - 1

    return False

if __name__ == "__main__":
    # Example input
    a = [1, 8, 4, 5, 6, 7, 16, 10, 11, 13, 15]
    k = 11

    # Check if the element is in the list
    result = is_find(a, k)
    print(f"Output is {result}")
